OpenSim/Simulation/Model/Geometry.h

```diff
@@ -611,7 +615,12 @@ class OSIMSIMULATION_API Mesh : public Geometry
     /// Method to map Mesh to Array of SimTK::DecorativeGeometry.
     void implementCreateDecorativeGeometry(
         SimTK::Array_<SimTK::DecorativeGeometry>& decoGeoms) const override;
-
+private:
+    // We cache the DecorativeMeshFile and a flag indicating if we tried to
+    // load the mesh from file so we don't try loading from disk every frame.
+    // These are mutable since they're not part of the public interface.
+    mutable std::shared_ptr<SimTK::DecorativeMeshFile> cachedMesh;
+    mutable bool triedToLoad;
```

Comment:
@chrisdembia The reason there is a separate flag is that if we try to fail once and fail then the pointer will be null and we'll try again with every frame making the drawing excessively slow even if there's only one missing or failed-to-load mesh file. That's why the flag is used to indicate that we "tried". I don't expect the objects with  loaded cachedMesh to be copies so I'm fine with either shared_ptr or unique_ptr.